src/karapace/forward_client.py

```diff
@@ -14,16 +16,30 @@
 LOG = logging.getLogger(__name__)
 
 
+T = TypeVar("T", bound=BaseModel)
+P = TypeVar("P", bound=Union[int, list[int]])
+
+
 class ForwardClient:
     USER_AGENT = f"Karapace/{__version__}"
 
-    def __init__(self):
+    def __init__(self) -> None:
         self._forward_client: aiohttp.ClientSession | None = None
 
     def _get_forward_client(self) -> aiohttp.ClientSession:
         return aiohttp.ClientSession(headers={"User-Agent": ForwardClient.USER_AGENT})
 
-    async def forward_request_remote(self, *, request: Request, primary_url: str) -> Response:
+    def _acceptable_response_content_type(self, *, content_type: str) -> bool:
+        return (
+            content_type.startswith("application/") and content_type.endswith("json")
+        ) or content_type == "application/octet-stream"
+
+    async def _forward_request_remote(
+        self,
+        *,
+        request: Request,
+        primary_url: str,
+    ) -> bytes:
         LOG.info("Forwarding %s request to remote url: %r since we're not the master", request.method, request.url)
         timeout = 60.0
         headers = request.headers.mutablecopy()
```

Comment:
There's also this piece of code: `func = getattr(self._get_forward_client(), request.method.lower())`, this would mean that we generate a new client every time we want to forward a request. Maybe we move the client initialization to the `__init__()` function, assign directly to `self._forward_client` and that will get reused for the whole app runtime and we can even add a `close()` function that cleans up the client session.

Reply:
Yes, probably not in this PR though.